src/utils/titleHelper.ts

```diff
@@ -1,19 +1,35 @@
 import { PageTitles } from 'common/constants';
 
+type EntityType =
+  | 'Project'
+  | 'Build Config'
+  | 'Group Config'
+  | 'Build'
+  | 'Group Build'
+  | 'Product'
+  | 'Version'
+  | 'Milestone'
+  | 'Artifact'
+  | 'SCM Repository'
+  | '';
+
 // TODO: Add appropriate type for serviceContainer
 interface IGeneratePageTitle {
   pageType?: 'Detail' | 'Edit' | 'Create';
   serviceContainer: any;
-  entity: 'Project' | 'Build Config' | 'Group Config' | 'Build' | 'Group Build' | 'Product' | 'Artifact' | 'SCM Repository';
+  firstLevelEntity: EntityType;
+  nestedEntity?: EntityType;
   entityName?: string;
 }
 
 export const generatePageTitle = ({
```

Comment:
Could you add 2 pattern based examples containing variant 1) with 2) without nested entity?

something like:
```tsx
/**
* example output 1: <nestedEntity> · <firstLevelEntity> <pageType> · PNC
* example output 2: ...
*/
```